src/wiki/templates/wiki/includes/revision_info.html

```diff
@@ -7,7 +7,7 @@
 
 
 {% load wiki_tags i18n %}
-{% if not hidedate %}{{ revision.created }}{% endif %} {% if not hidenumber %}(#{{ revision.revision_number }}) {% trans "by" %}{% endif %} {% if revision.user %}{{ revision.user }}{% else %}{% if article|can_moderate:user %}{{ revision.ip_address|default:"anonymous (IP not logged)" }}{% else %}{% trans "anonymous (IP logged)" %}{% endif %}{% endif %}
+{% if not hidedate %}{{ revision.created }}{% endif %} {% if not hidenumber %}(#{{ revision.revision_number }}) {% trans "by" %}{% endif %} {% if revision.user %}{{ revision.user }}{% else %}{% if article|can_moderate:user or not revision.ip_address %}{{ revision.ip_address|default:"anonymous (IP not logged)" }}{% else %}{% trans "anonymous (IP logged)" %}{% endif %}{% endif %}
```

Comment:
```suggestion
{% if not hidedate %}{{ revision.created }}{% endif %}

{% if not hidenumber %}(#{{ revision.revision_number }}) {% trans "by" %}{% endif %}

{% if revision.user %}
  {{ revision.user }}
{% else %}
{% if not revision.ip_address %}
  {% trans "anonymous (IP not logged)" %}
{% else %}
  {% if article|can_moderate:user %}
    {{ revision.ip_address }}
  {% else %}
    {% trans "anonymous (IP logged)" %}
  {% endif %}
{% endif %}
```

This is admittedly a very unreadable line and I think that since it's HTML we're allowed to add linebreaks without changing the final result :)

We can also shuffle around the logic a bit.

If you like it, can you check and see?